fix(Makefile): more specific path for make rules

Henrik66 opened the pull request:
## Checklist
- [X] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Henrik66 committed:
fix(Makefile): more specific path for make rules

LaszloGombos reviewed (approved):
LGTM

aafeijoo-suse reviewed (commented):
Are you sure we are not building `util.o` using `CC` and `LDFLAGS`? Please, correct me if I'm wrong, maybe I need to refresh my Makefile bits.

```
    # make -j4
    cc -c -O2 -g -Wall -std=gnu99 -D_FILE_OFFSET_BITS=64 -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2  -I/usr/include/kmod src/install/dracut-install.c -o src/install/dracut-install.o
    cc -c -O2 -g -Wall -std=gnu99 -D_FILE_OFFSET_BITS=64 -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2  -I/usr/include/kmod src/install/hashmap.c -o src/install/hashmap.o
    cc -c -O2 -g -Wall -std=gnu99 -D_FILE_OFFSET_BITS=64 -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2  -I/usr/include/kmod src/install/log.c -o src/install/log.o
    cc -c -O2 -g -Wall -std=gnu99 -D_FILE_OFFSET_BITS=64 -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2  -I/usr/include/kmod src/install/strv.c -o src/install/strv.o
    cc -c -O2 -g -Wall -std=gnu99 -D_FILE_OFFSET_BITS=64 -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2  -I/usr/include/kmod src/install/util.c -o src/install/util.o
    cc -c -O2 -g -Wall -std=gnu99 -D_FILE_OFFSET_BITS=64 -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2  -I/usr/include/kmod src/skipcpio/skipcpio.c -o src/skipcpio/skipcpio.o
--> cc -c -O2 -g -Wall -std=gnu99 -D_FILE_OFFSET_BITS=64 -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2  -I/usr/include/kmod src/util/util.c -o src/util/util.o
    cc   src/skipcpio/skipcpio.o   -o src/skipcpio/skipcpio
    cc   src/util/util.o   -o src/util/util
    cp -a src/util/util dracut-util
    cc  -o src/install/dracut-install src/install/dracut-install.o src/install/hashmap.o src/install/log.o src/install/strv.o src/install/util.o  -lc -lkmod
    ln -fs src/install/dracut-install dracut-install
```

Henrik66 commented:
I changed the PR based on @LaszloGombos suggestion, but I removed the reference to bug 2366 as it seems we do not have a way to reproduce or understand that bug better.
